<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>QIIME2</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Microbiome Hub</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Getting_Started.html">Getting Started</a>
</li>
<li>
  <a href="Unix_Linux_Cheat_Sheet.html">Unix/Linux Cheat Sheet </a>
</li>
<li>
  <a href="Data_Download.html">Data Download</a>
</li>
<li>
  <a href="QIIME2.html">QIIME2</a>
</li>
<li>
  <a href="Glossary.html">Glossary</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">QIIME2</h1>

</div>


<p>This tutorial will take you through the basics of using QIIME2 with data from UC San Diego.</p>
<div id="import-and-demultiplex-data" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Import and Demultiplex Data</h1>
<p>We are going to create an sbatch file with commands for importing and demultiplexing data using QIIME2.</p>
<div id="create-an-sbatch-file." class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Create an SBATCH File.</h2>
<p>First, let’s open a text editor like nano where you can edit your document:</p>
<pre class="markdown"><code>nano QIIME2_Import_Demux.sbatch</code></pre>
<p>You should now have an empty file called QIIME2_Import_Demux.sbatch. We are going to create an SBATCH file that uses QIIME2 for these steps:</p>
<pre class="markdown"><code>#!/bin/bash
#SBATCH -J 16S_Import_Demux
#SBATCH -o log.o%j 
#SBATCH -p bsudfq
#SBATCH -N 1
#SBATCH -n 14
#SBATCH -t 1-00:00:00
#SBATCH --mail-type=END
#SBATCH --mail-type=BEGIN
#SBATCH --mail-type=FAIL
#SBATCH --mail-user=user@domain.edu

# Ensure the temporary directory is in your scratch, so that big files don&#39;t take up too much space.

export TMPDIR=~/scratch

# You can create a &#39;shortcut&#39; to the folder you&#39;re going to be working out of:

16S=~/scratch/16S

# Input. To analyze these data, the sequences must first be imported into an artifact of type EMPPairedEndSequences. You want to ensure that your input-path is a folder that has your forward reads, reverse reads, and barcodes only.

qiime tools import \
  --type EMPPairedEndSequences \
  --input-path ${16S}/Reads \
  --output-path ${16S}/emp-paired-end-sequences.qza

# Demux. In order to demultiplex your data, you will need an additional barcode file that came with your data. You will want a folder called &#39;demux_output&#39; where your outputs can go. 

mkdir ${16S}/demux_output

qiime demux emp-paired \
    --m-barcodes-file ${16S}/110599_mapping_fileRename.tsv \
    --m-barcodes-column BarcodeSequence \
    --p-rev-comp-barcodes \
    --p-rev-comp-mapping-barcodes \
    --i-seqs ${16S}/emp-paired-end-sequences.qza \
    --o-per-sample-sequences ${16S}/demux_output/demux.qza \
    --o-error-correction-details ${16S}/demux_output/demux-details.qza

# Summarize. This will summarize our data with visualization files (*.qzv) in QIIME2:

qiime demux summarize \
  --i-data ${16S}/demux_output/demux.qza \
  --o-visualization ${16S}/demux_output/per_sample_sequences.qzv
</code></pre>
<p>Once you have sorted your file paths that are specific to your configuration on the HPC, you can save this file by pressing control + x. Press ‘y’ to save your file.</p>
</div>
<div id="run-the-sbatch-file-on-your-hpc." class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Run the SBATCH file on your HPC.</h2>
<p>Now you’re ready to run your first SBATCH file on Borah. Let’s use the following commands:</p>
<pre class="markdown"><code># Ensure that you call conda activate qiime2-2021.4 before you start. 
conda activate qiime2-2021.4

# Submit your job:
sbatch QIIME2_Import_Demux.sbatch

# Check in on your job:
squeue

# Cancel your job if necessary
scancel &lt;jobID&gt;
</code></pre>
<p>If your job fails, take a look at your log file using the ‘cat’ function. The first error message should indicate what went wrong.</p>
<pre class="markdown"><code>cat log.o&lt;jobID&gt;</code></pre>
<p>If successful, this import should have created QIIME2 artifact files (.qza) and QIIME2 visualization files (.qzv).</p>
</div>
<div id="visualize-the-data." class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Visualize the data.</h2>
<p>You will want to import the visualization files from the HPC to your local (personal) computer using the following commands:</p>
<pre class="markdown"><code>#In a terminal on your personal computer:
mkdir ~/16S
cd ~/16S

scp login@borah-login.boisestate.edu:~/scratch/16S/demux_output/per_sample_sequences.qzv ./</code></pre>
<p>Once per_sample_sequences.qsv is on your local computer, you can visualize it. To do this, go to <a href="https://view.qiime2.org/" class="uri">https://view.qiime2.org/</a>. Drag your file to this site to show the visualization.</p>
<p>The first page you will see will give an overview of descriptive statistics for your sequencing data, including read counts.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:DescStats"></span>
<img src="Images/Descriptive_Stats.png" alt="Read count descriptive statistics from pronghorn data set." width="100%" />
<p class="caption">
Figure 1.1: Read count descriptive statistics from pronghorn data set.
</p>
</div>
<p>These will also be provided in a histogram to show the distribution of read counts.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:histo"></span>
<img src="Images/Read_Count_Histogram.png" alt="Read count histogram from pronghorn data set." width="100%" />
<p class="caption">
Figure 1.2: Read count histogram from pronghorn data set.
</p>
</div>
<p>If you navigate to the ‘Interactive Quality Plot’ tab, you will find quality scores for all of our sequence reads, which will be used to trim for DADA2 pipeline. This plot is created from a subsample of 10,000 reads. This number is the default parameter for the demux plugin, but can be adjusted.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:qual"></span>
<img src="Images/Interactive_Quality_Score.png" alt="Interactive quality plot from pronghorn data set." width="100%" />
<p class="caption">
Figure 1.3: Interactive quality plot from pronghorn data set.
</p>
</div>
<p>The quality scores at the ends of each direction tell us how we should trim the data during the DADA2 denoising step. The noisier the data, the more likely it is that low abundance variation is mistaken for error and reads fail to merge. Overall, the quality scores of this data set are pretty high (&gt;30). We can see that quality score improves in both directions after 12bp (the length of the barcode adapters from UCSD). We also see quality drop a bit at 151bp (the length of the reads from a MiSeq). We therefore will truncate our data accordingly in the next step.</p>
</div>
</div>
<div id="dada2" class="section level1" number="2">
<h1><span class="header-section-number">2</span> DADA2</h1>
<p>This method denoises paired-end sequences, dereplicates them, and filters chimeras (i.e., artifacts from the PCR process from 2+ biological sequences incorrectly joined together). In the dereplication process all the quality-filtered sequences are collapsed into a set of unique reads, making downstream computation easier.</p>
<div id="sbatch-for-running-dada2" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> SBatch for Running DADA2</h2>
<p>Let’s open a text editor like nano where you can edit your document:</p>
<pre class="markdown"><code>nano QIIME2_DADA2.sbatch</code></pre>
<p>You should now have an empty file called QIIME2_DADA2.sbatch. We are going to create an SBATCH file that uses QIIME2 for these steps:</p>
<pre class="markdown"><code>#!/bin/bash
#SBATCH -J 16S_DADA2
#SBATCH -o log.o%j 
#SBATCH -p bsudfq
#SBATCH -N 1
#SBATCH -n 14
#SBATCH -t 1-00:00:00
#SBATCH --mail-type=END
#SBATCH --mail-type=BEGIN
#SBATCH --mail-type=FAIL
#SBATCH --mail-user=user@domain.edu

# Ensure the temporary directory is in your scratch, so that big files don&#39;t take up too much space.

export TMPDIR=~/scratch

# You can create a &#39;shortcut&#39; to the folder you&#39;re going to be working out of:

16S=~/scratch/16S

# We are now going to denoise the data, cutting off the first 12 base pairs. 
qiime dada2 denoise-paired \
  --i-demultiplexed-seqs ${16S}/demux_output/demux.qza \
  --p-trim-left-f 12 \
  --p-trim-left-r 12 \
  --p-trunc-len-f 150 \
  --p-trunc-len-r 150 \
  --p-n-threads 10 \
  --o-representative-sequences ${16S}/demux_output/rep-seqs-dada2.qza \
  --o-table ${16S}/demux_output/table-dada2.qza \
  --o-denoising-stats ${16S}/demux_output/stats-dada2.qza

# Feature table
# Here, make sure to use your metadata file, as opposed to your barcode file. 
qiime feature-table summarize   \
  --i-table ${16S}/demux_output/table-dada2.qza \
  --o-visualization ${116S}/demux_output/table-dada2.qzv \
  --m-sample-metadata-file ${16S}/12581_20210414-100355.tsv

# Representative sequences
qiime feature-table tabulate-seqs \
  --i-data ${16S}/demux_output/rep-seqs-dada2.qza \
  --o-visualization ${16S}/demux_output/rep-seqs-dada2.qzv

# Denoising Stats
qiime metadata tabulate \
  --m-input-file ${16S}/demux_output/stats-dada2.qza \
  --o-visualization ${16S}/demux_output/stats-dada2.qzv

## Export ASV table

qiime tools export \
  --input-path ${16S}/demux_output/table-dada2.qza \
  --output-path ${16S}/exported-files

biom convert -i ${16S}/exported-files/feature-table.biom \
-o $16S}/exported-files/asv-table-dada2.txt \
--to-tsv </code></pre>
<p>Once you have sorted your file paths that are specific to your configuration on the HPC, you can save this file by pressing control + x. Press ‘y’ to save your file.</p>
</div>
<div id="run-the-sbatch-file-on-your-hpc.-1" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Run the SBATCH file on your HPC.</h2>
<p>Now you’re ready to run your second SBATCH file on Borah. Let’s use the following commands:</p>
<pre class="markdown"><code># Ensure that you call conda activate qiime2-2021.4 before you start. 
conda activate qiime2-2021.4

# Submit your job:
sbatch QIIME2_DADA2.sbatch

# Check in on your job:
squeue

# Cancel your job if necessary
scancel &lt;jobID&gt;
</code></pre>
<p>If your job fails, take a look at your log file using the ‘cat’ function. The first error message should indicate what went wrong.</p>
<pre class="markdown"><code>cat log.o&lt;jobID&gt;</code></pre>
<p>If successful, this import should have created QIIME2 artifact files (.qza) and QIIME2 visualization files (.qzv).</p>
</div>
<div id="visualize-the-data.-1" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Visualize the data.</h2>
<p>You will want to import the visualization files from the HPC to your local (personal) computer using the following commands:</p>
<pre class="markdown"><code>#In a terminal on your personal computer:
mkdir ~/16S
cd ~/16S

scp login@borah-login.boisestate.edu:~/scratch/16S/demux_output/per_sample_sequences.qzv ./</code></pre>
<p>Once per_sample_sequences.qsv is on your local computer, you can visualize it. To do this, go to <a href="https://view.qiime2.org/" class="uri">https://view.qiime2.org/</a>. Drag your file to this site to show the visualization.</p>
</div>
</div>
<div id="taxonomy" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Taxonomy</h1>
<p>In order to assign taxonomy, we use pre-existing SILVA classifiers from the QIIME2 website: <a href="https://docs.qiime2.org/2021.2/data-resources/" class="uri">https://docs.qiime2.org/2021.2/data-resources/</a></p>
<p>This includes:
Silva 138 SSURef NR99 515F/806R region sequences
Silva 138 SSURef NR99 515F/806R region taxonomy
Silva 128 SEPP reference database</p>
<p>These can be downloaded directly to the cluster using the following commands. Make sure that you’re in the correct database.</p>
<pre class="markdown"><code>cd ~/scratch/16S/
wget https://data.qiime2.org/2021.2/common/silva-138-99-seqs-515-806.qza
wget https://data.qiime2.org/2021.2/common/silva-138-99-tax-515-806.qza
wget https://data.qiime2.org/2021.2/common/sepp-refs-silva-128.qza</code></pre>
<div id="sbatch-for-taxonomy" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> SBatch for Taxonomy</h2>
<p>Let’s open a text editor like nano where you can edit your document:</p>
<pre class="markdown"><code>nano QIIME2_DADA2.sbatch</code></pre>
<p>You should now have an empty file called QIIME2_DADA2.sbatch. We are going to create an SBATCH file that uses QIIME2 for these steps:</p>
<p>```markdown
#!/bin/bash
#SBATCH -J 16S_Taxonomy
#SBATCH -o log.o%j
#SBATCH -p bsudfq
#SBATCH -N 1
#SBATCH -n 14
#SBATCH -t 1-00:00:00
#SBATCH –mail-type=END
#SBATCH –mail-type=BEGIN
#SBATCH –mail-type=FAIL
#SBATCH –<a href="mailto:mail-user=user@domain.edu" class="email">mail-user=user@domain.edu</a></p>
</div>
</div>
<div id="ensure-the-temporary-directory-is-in-your-scratch-so-that-big-files-dont-take-up-too-much-space." class="section level1" number="4">
<h1><span class="header-section-number">4</span> Ensure the temporary directory is in your scratch, so that big files don’t take up too much space.</h1>
<p>export TMPDIR=~/scratch</p>
</div>
<div id="you-can-create-a-shortcut-to-the-folder-youre-going-to-be-working-out-of" class="section level1" number="5">
<h1><span class="header-section-number">5</span> You can create a ‘shortcut’ to the folder you’re going to be working out of:</h1>
<p>16S=~/scratch/16S</p>
</div>
<div id="taxonomy-train-the-classifier" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Taxonomy: Train the classifier</h1>
<p>qiime feature-classifier fit-classifier-naive-bayes<br />
–i-reference-reads ${Sage_Grouse_16S}/silva-138-99-seqs-515-806.qza<br />
–i-reference-taxonomy ${Sage_Grouse_16S}/silva-138-99-tax-515-806.qza<br />
–o-classifier ${Sage_Grouse_16S}/classifier.qza</p>
<p>#Phylogenetic Tree
qiime fragment-insertion sepp<br />
–i-representative-sequences ${Sage_Grouse_16S}/demux_output/rep-seqs-dada2.qza<br />
–i-reference-database ${Sage_Grouse_16S}/sepp-refs-silva-128.qza<br />
–p-threads 4<br />
–o-tree ${Sage_Grouse_16S}/sepp-tree.qza<br />
–o-placements ${Sage_Grouse_16S}/sepp-placements.qza</p>
<p>qiime feature-classifier classify-sklearn<br />
–i-reads ${Sage_Grouse_16S}/demux_output/rep-seqs-dada2.qza<br />
–i-classifier ${Sage_Grouse_16S}/classifier.qza<br />
–o-classification ${Sage_Grouse_16S}/taxonomy.qza</p>
<p>qiime metadata tabulate<br />
–m-input-file ${Sage_Grouse_16S}/taxonomy.qza<br />
–o-visualization ${Sage_Grouse_16S}/taxonomy.qzv</p>
<p>qiime taxa barplot<br />
–i-table ${Sage_Grouse_16S}/demux_output/table-dada2.qza<br />
–i-taxonomy ${Sage_Grouse_16S}/taxonomy.qza<br />
–m-metadata-file ${Sage_Grouse_16S}/12842_20210216-155838.tsv<br />
–o-visualization ${Sage_Grouse_16S}/taxa-bar-plots.qzv</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
